### ISSUE TITLE
Codespaces extension tries (and fails) to log in on every window

### ISSUE THREAD

**Issue by connor4312** [MicrosoftDocs/vsonline#634]
A few minutes ago it seems like Codespaces wants to sign in again. (also noticed that @eamodio's machine started the same thing on a screenshare at the same time)

![](https://memes.peet.io/img/20-07-ea1f4d88-ec1b-4922-acdb-32178d012e7f.png)

I tried to do so, and ended up on an auth screen that never let me continue:

https://memes.peet.io/img/20-07-3fffb92b-d3b1-43d3-95dd-f5fc1d9cf70b.webm

It seems the only solution to all new code windows opening a browser is to disable the codespaces extension.

**Comment by cmbrose** [comment]
Hi @connor4312!  I can't quite tell from the image - are you seeing this in desktop or in browser vs code?  Could you also share your extension and vs code version info?  Thanks!

**Comment by connor4312** [comment]
Desktop VS Code with codespaces 1.0.2427.

```
Version: 1.47.0-insider (user setup)
Commit: 04545fa88043fd10d1f3edefd26be1b8245b516f
Date: 2020-07-02T05:48:37.715Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

I'm still able to reproduce this this morning.

**Comment by gregvanl** [comment]
Seeing this as well whenever I restart VS Code Insiders. I don't see this on VS Code Stable 1.46.1.

**Comment by sheldonhull** [comment]
Added my vote, but additionally wanted to add that if I have vscode set to open all prior folders, that it results in opening this [login page](https://login.microsoftonline.com/common "login page for vsonline") multiple times. My system was not happy 😁

**Comment by anthonychu** [comment]
Same here. I'm on macOS and VS Code stable. /cc @legomushroom

**Comment by legomushroom** [comment]
I believe VSCode had regression with the keytar recently but thought that is fixed already, LiveShare team hit the same issue a week ago, @jramsay did you follow up on this?

**Comment by jramsay** [comment]
@legomushroom: here is this issue we hit a few weeks ago in Live Share. It had an impact on desktop & web. Fixed by VS Code in 1.46.1:  
[https://github.com/microsoft/vscode/issues/100333 ](https://github.com/microsoft/vscode/issues/100333 )

**Comment by legomushroom** [comment]
@anthonychu where is the VSCode instance installed on your mac? Is it in the applications folder?

**Comment by anthonychu** [comment]
@legomushroom Yes. Both stable (1.47.0) and insiders (1.48.0) are, and both have the behavior.

**Comment by akoeplinger** [comment]
I'm hitting the same issue however in my case I get past the authentication prompt in the browser and pop back into VSCode, it just seems the extension never finishes signing in.

```
Version: 1.47.2
Commit: 17299e413d5590b14ab0340ea477cdd86ff13daf
Date: 2020-07-15T18:18:50.054Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.5.0
```

VSCode extension version: `v1.0.2471`

Ìnstalled into /Applications, using MS Edge as the browser.

My Microsoft alias is `alkpli` if you want to debug this internally.
